### ISSUE TITLE
Search in combo for text with spaces is broken

### ISSUE THREAD

**Issue by Chrriis** [JFormDesigner/FlatLaf#691]
When a combo has the focus (with or without the popup showing), typing text moves the selection to the item matching what is being typed. Unfortunately with FlatLaf this does not work when the item has a space, because hitting space shows or hides the popup.

Searching with keyboard items with a space works with the Windows look and feel.

Here is a simple test case showing the issue:
[ComboBoxTest.zip](https://github.com/JFormDesigner/FlatLaf/files/11747974/ComboBoxTest.zip)

Just launch, and try to select with the keyboard, for example, "d z":
![image](https://github.com/JFormDesigner/FlatLaf/assets/1174202/0488251f-51d1-4e70-a51f-963baa37ca66)

**Comment by Chrriis** [comment]
The fix for the issue is simply to remove that line:
https://github.com/JFormDesigner/FlatLaf/blob/6f6a860887de0896b817a53140a830784d0e548c/flatlaf-core/src/main/java/com/formdev/flatlaf/FlatInputMaps.java#L53

**Comment by DevCharly** [comment]
This is actually a bug in Swing.

Metal, Nimbus and Aqua L&F have the same problem because they also use the `SPACE` key to show the popup.

Removing the `SPACE` key mapping would fix this issue, but then it is no longer possible to show the popup using the Space key. Not sure how important that is...

In native macOS apps, the Space key shows the combobox popup.
Also WinUI apps (Windows 10/11) show the combobox popup when pressing the Space key.
But Win32 apps do not...

I'll check whether it is possible to fix this without removing the  `SPACE` key mapping...

If not, then we either

- loose the "Space key shows popup" functionality
- or we live with this bug (and hope that it is fixed in Swing)

**Comment by Chrriis** [comment]
On Windows look and feel, "down" moves to the next item, "alt+down" shows the popup.

This search in combo boxes with spaces was reported by some users who had long list of things (people with firstname lastname, some objects with the same first keyword, etc.) and were complaining about the regression with the move to FlatLaf.
I had no choice but to patch the FlatLaf library so I cannot use the standard release anymore...

Do you know if there is an alternative way to remove that mapping without patching the library?

**Comment by DevCharly** [comment]
> Do you know if there is an alternative way to remove that mapping without patching the library?

Sure:

~~~java
UIManager.setLookAndFeel(new FlatLightLaf());

// remove Space key mapping from shared ComboBox input map, which is used for all combo boxes
InputMap inputMap = (InputMap) UIManager.get( "ComboBox.ancestorInputMap" );
if( inputMap != null )
    inputMap.remove( KeyStroke.getKeyStroke( "SPACE" ) );
~~~

**Comment by Chrriis** [comment]
Thanks for the sample code!

I tried on a combo box in Windows 10 settings and the popup is shown when pressing space only if not currently searching for an item:
- "&lt;space&gt;": opens popup.
- "John&lt;space&gt;": does not open the popup.

I wonder if there is a way in Swing to block the keystroke action if currently in search mode.

For the time being, I removed the space key stroke using the code you suggested.
